vm/systemSmartContracts/staking.go

```diff
@@ -200,14 +200,17 @@ func (r *stakingSC) removeFromJailedNodes() {
 	r.setConfig(config)
 }
 
-func (r *stakingSC) canUnStake() bool {
+func (r *stakingSC) numSpareNodes() int64 {
 	config := r.getConfig()
-	return config.StakedNodes-config.JailedNodes-config.MinNumNodes > 0
+	return config.StakedNodes - config.JailedNodes - config.MinNumNodes
+}
+
+func (r *stakingSC) canUnStake() bool {
+	return r.numSpareNodes() > 0
 }
 
 func (r *stakingSC) canUnBond() bool {
-	config := r.getConfig()
-	return config.StakedNodes-config.JailedNodes-config.MinNumNodes >= 0
+	return r.numSpareNodes() >= 0
```

Comment:
should not be checked with > 0 ?

Reply:
no. in case of unbond even it is equal he can leave :) as it means there are enough in eligible list. in case of unstake you do not let it, because if someone unstakes the number of eligible gets under the minimum barrier. When unbond, he already did the unstake part.

Reply:
ok